price-estimator/src/filter.rs

```diff
@@ -202,8 +203,9 @@ async fn estimate_buy_amount<T>(
         sell_amount_in_quote * token_info.base_unit_in_atoms()
     };
     let transitive_order = orderbook
-        .get_pricegraph()
+        .get_pricegraph(query.batch_id)
         .await
+        .map_err(error::internal_server_rejection)?
```

Comment:
What error is displayed when we query a future batch?

Reply:
I don't think it is an error if we query a future batch, but this can be implemented.

Reply:
Agree this can be a separate PR. I would personally prefer a tighter API that yells at you if you are doing something that doesn't make sense.